devito/at_controller.py

```diff
@@ -0,0 +1,344 @@
+                self._set_x_permission(self._make_f_path)
+
+    def auto_tune(self, file_path, time_order, space_order):
+        """Does the auto tuning for selected file and collects relevant data
+
+        Args:
+            file_path (str): full path to the file that we want to auto tune
+            time_order (int): time order of kernel.
+            space_order (int): space order of kernel.
+        """
+
+        try:
+            print "Starting Auto tuning for %s" % file_path
+            self.time_order = time_order
+            self.space_order = space_order
+
+            # these functions have to run in this order
+            self._prep_file_for_at(file_path)
+            self._run_at()
+            self._extract_report_info()
+            print "Auto tuning  complete for %s" % file_path
+        except ValueError:
+            print "Auto tuning for %s failed" % file_path
+
+        self._at_parent_folder = ""  # reset parent folder name
+
+    def _prep_file_for_at(self, file_path):
+        """Copies file to work dir and copies all at isat-files there
+
+        Args:
+            file_path (str): full path to the file that we want to auto tune
+
+        Raises:
+            ValueError: if file at file_path is not found
+        """
+
+        if not os.path.isfile(file_path):
+            print "%s not found"
+            raise ValueError()
+
+        self._at_parent_folder = os.path.join(self._at_work_dir, "time_o_%d_space_o_%d" %
+                                              (self.time_order, self.space_order))
+        src_dir_path = os.path.join(self._at_parent_folder, self._at_src_dir)
+
+        # prep the parent folder
+        if not os.path.isdir(self._at_parent_folder):
+            os.mkdir(self._at_parent_folder)
+        else:
+            self._clean_folder(self._at_parent_folder)
+
+        # create src and dst folders which will be used for at
+        os.mkdir(src_dir_path)
+
+        # copy file which will be at to src dir
+        shutil.copyfile(file_path, os.path.join(src_dir_path, "%s.cpp" % self._at_f_name))
+
+        # copy required isat files
+        shutil.copy2(self._build_f_path, os.path.join(src_dir_path, self._isat_build_f))
+        shutil.copy2(self._clean_f_path, os.path.join(src_dir_path, self._isat_clean_f))
+        shutil.copy2(self._test_f_path, os.path.join(src_dir_path, self._isat_test_f))
+        shutil.copy2(self._make_f_path, os.path.join(src_dir_path, self._make_f))
+
+    def _run_at(self):
+        """Runs isat.py which does the auto tuning
+
+        Raises:
+            ValueError: If isat.py return code != 0
+        """
+
+        log_file_path = os.path.join(self._at_parent_folder, "log.txt")  # Writes STDOUT in here
+        log = open(log_file_path, 'w')
+
+        # auto tuning command. -i for source of at -o for destination of at result
+        cmd = "%s -i %s -o %s" % (self._isat_py_path, os.path.join(self._at_parent_folder, self._at_src_dir, ),
```

Comment:
I meant that I just redirected std out into log file. Isat does not write log itself. 
In any case i tried few different ways however its constantly unhappy with import dependencies in isat.  Don't want to waste time, on this anymore. If it's really bothering you please suggest me they way to solve this, so I wouldn't get tons of import errors due to depenedencies